src/main/java/org/kohsuke/groovy/sandbox/impl/Checker.java

```diff
@@ -573,6 +596,194 @@ public static Object checkedBinaryOp(Object lhs, int op, Object rhs) throws Thro
+     *
+     * @see UnaryExpressionHelper#writeBitwiseNegate
+     * @see ScriptBytecodeAdapter#bitwiseNegate
+     * @see InvokerHelper#bitwiseNegate
+     */
+    public static Object checkedBitwiseNegate(Object value) throws Throwable {
+        if (value instanceof Integer) {
+            return ~((Integer)value);
+        }
+        if (value instanceof Long) {
+            return ~((Long)value);
+        }
+        if (value instanceof BigInteger) {
+            return checkedCall(value, false, false, "not", new Object[]{});
+        }
+        if (value instanceof String) {
+            // value is a regular expression.
+            return checkedStaticCall(StringGroovyMethods.class, "bitwiseNegate", new Object[]{ value.toString() });
+        }
+        if (value instanceof GString) {
+            // value is a regular expression.
+            return checkedStaticCall(StringGroovyMethods.class, "bitwiseNegate", new Object[]{ value.toString() });
+        }
+        if (value instanceof ArrayList) { // ArrayList is the exact type that Groovy checks in InvokerHelper.bitwiseNegate.
+            // value is a list.
+            List newlist = new ArrayList();
+            for (Object element : ((ArrayList) value)) {
+                newlist.add(checkedBitwiseNegate(element));
+            }
+            return newlist;
+        }
+        return checkedCall(value, false, false, "bitwiseNegate", EMPTY_ARRAY);
+    }
+
+    /**
+     * Intercepts range expressions of the form {@code [x..y]} or {@code [x..<y]}.
+     *
+     * If the from and to expressions are not integers, this operator constructs a {@link ObjectRange}, which involves
+     * calling various methods on the endpoints during the construction of the range and whenever the range is used.
+     * We handle this like we do interfaces in {@link #preCheckedCast} and intercept all of the methods that may be
+     * called before allowing the range to be constructed rather than trying to implement our own range type that is
+     * sandbox-aware.
+     *
+     * @see ScriptBytecodeAdapter#createRange
+     * @see ObjectRange
+     */
+    public static Object checkedCreateRange(Object from, Object to, boolean inclusive) throws Throwable {
+        if (from instanceof Integer && to instanceof Integer) {
+            if (inclusive || from != to) {
+                return checkedConstructor(IntRange.class, new Object[] { inclusive, from, to });
+            }
+            // Fallthrough for EmptyRange
+        }
+        if (!inclusive) {
+            if (Boolean.TRUE.equals(checkedComparison(from, Types.COMPARE_EQUAL, to))) {
+                // Unchecked cast to Comparable matches the behavior of ScriptBytecodeAdapter.createRange.
+                return checkedConstructor(EmptyRange.class, new Object[] { (Comparable)from });
+            }
+            if (checkedComparison(from, Types.COMPARE_GREATER_THAN, to) == Boolean.TRUE) {
```

Comment:
```suggestion
            if (Boolean.TRUE.equals(checkedComparison(from, Types.COMPARE_GREATER_THAN, to))) {
```
to get SpotBugs to shut up

Reply:
Oops yeah I forgot to push this change earlier.